tests/mypy_test.py

```diff
@@ -361,14 +371,36 @@ def test_stdlib(code: int, args: TestConfig) -> TestResults:
         if module_min_version <= tuple(map(int, args.version.split("."))) <= module_max_version:
             add_files(files, (stdlib / name), args)
 
-    if files:
-        print(f"Testing stdlib ({len(files)} files)...", end="", flush=True)
-        # We don't actually need pip for the stdlib testing
-        venv_info = VenvInfo(pip_exe="", python_exe=sys.executable)
-        this_code = run_mypy(args, [], files, venv_info=venv_info, testing_stdlib=True, non_types_dependencies=False)
-        code = max(code, this_code)
+    if not files:
+        return TestResult(MypyResult.SUCCESS, 0)
+
+    print(f"Testing stdlib ({len(files)} files)...", end="", flush=True)
+    # We don't actually need pip for the stdlib testing
+    venv_info = VenvInfo(pip_exe="", python_exe=sys.executable)
+    result = run_mypy(args, [], files, venv_info=venv_info, testing_stdlib=True, non_types_dependencies=False)
+    return TestResult(result, len(files))
+
 
-    return TestResults(code, len(files))
+@dataclass(frozen=True)
+class TestSummary:
+    mypy_result: MypyResult = MypyResult.SUCCESS
+    files_checked: int = 0
+    packages_skipped: int = 0
+    packages_with_errors: int = 0
+
+    def register_result(self, mypy_result: MypyResult, files_checked: int) -> TestSummary:
+        errors = 1 if mypy_result != MypyResult.SUCCESS else 0
+        return self.combine(TestSummary(mypy_result, files_checked, 0, errors))
+
+    def skip_package(self) -> TestSummary:
+        return TestSummary(self.mypy_result, self.files_checked, self.packages_skipped + 1)
+
+    def combine(self, other: TestSummary) -> TestSummary:
+        result = self.mypy_result if self.mypy_result.value > other.mypy_result.value else other.mypy_result
+        checked = self.files_checked + other.files_checked
+        skipped = self.packages_skipped + other.packages_skipped
+        errors = self.packages_with_errors + other.packages_with_errors
+        return TestSummary(result, checked, skipped, errors)
```

Comment:
Maybe the dataclass should be non-frozen? Then these methods could update the dataclass in-place, and return `None`? No strong opinion, though

Reply:
I am trying to avoid having to pass the previous summaries into the various functions so they are a bit more "standalone". Not particular necessary in our code, but I found this a bit cleaner.

Reply:
Although you are right, for the register/skip method this makes sense.

Reply:
Give me a moment, I'll come up with a suggestion

Reply:
I was mistaken anyway. Unfreezing it has no effect on passing it to functions.